src/libphonenumber/PhoneNumber.php

```diff
@@ -546,13 +546,17 @@ public function unserialize($serialized)
     {
         $data = unserialize($serialized);
 
-        $this->setCountryCode($data[0]);
-        $this->setNationalNumber($data[1]);
-        $this->setExtension($data[2]);
-        $this->setItalianLeadingZero($data[3]);
-        $this->setNumberOfLeadingZeros($data[4]);
-        $this->setRawInput($data[5]);
-        $this->setCountryCodeSource($data[6]);
-        $this->setPreferredDomesticCarrierCode($data[7]);
+        list (
+            $this->countryCode,
+            $this->nationalNumber,
+            $this->extension,
+            $this->italianLeadingZero,
+            $this->numberOfLeadingZeros,
+            $this->rawInput,
+            $this->countryCodeSource,
+            $this->preferredDomesticCarrierCode
+        ) = $data;
+
+        $this->hasNumberOfLeadingZeros = true;
```

Comment:
Note that this is consistent with the current behaviour: `unserialize` unconditionally called `setNumberOfLeadingZeros()`, which set this variable to `true`.

I think the current behaviour could be wrong though, and I would suggest serializing `hasNumberOfLeadingZeros` as well. Again, I can open another PR for this, but note that it would be a minor BC break if someone has stored serialized values with a previous version of `libphonenumber`, and unserializes them with a newer version.